src/main/java/com/cburch/logisim/file/Loader.java

```diff
@@ -142,6 +149,28 @@ private static File determineBackupName(File base) {
     return null;
   }
 
+  // Determine the autosave file name for any .circ file as .<basename>.circ.autosave
+  private static File determineAutosaveName(File base) {
+    if (base == null) {
+      String timestamp =
+          DateTimeFormatter.ofPattern("yyyy-MM-dd_HH:mm:ss").format(LocalDateTime.now());
```

Comment:
`:` in file names are not universally allowed, e.g., Windows doesn't allow them. Personally, I would opt to use `yyyyMMddHHmmss`, which also enables chronological sorting.